soundata/datasets/mavd.py

```diff
@@ -0,0 +1,303 @@
+        clip_id (str): clip id
+
+    """
+
+    def __init__(
+        self,
+        clip_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            clip_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.audio_path = self.get_path("audio")
+        self.txt_path = self.get_path("txt")
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The clip's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    @property
+    def split(self):
+        return self._clip_metadata.get("split")
+
+    @core.cached_property
+    def events(self) -> Optional[annotations.Events]:
+        return load_events(self.csv_path)
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO, sr=None) -> Tuple[np.ndarray, float]:
+    """Load a MAVD-traffic audio file.
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+        sr (int or None): sample rate for loaded audio, None by default, which
+            uses the file's original sample rate of 44100 without resampling.
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+
+    """
+    audio, sr = librosa.load(fhandle, sr=sr, mono=True)
+    return audio, sr
+
+
+@io.coerce_to_string_io
+def load_events(fhandle: TextIO) -> annotations.Events:
+    """Load an MAVD-traffic sound events annotation file
+    Args:
+        fhandle (str or file-like): File-like object or path to the sound events annotation file
+    Raises:
+        IOError: if txt_path doesn't exist
+    Returns:
+        Events: sound events annotation data
+    """
+
+    times = []
+    labels = []
+    confidence = []
+    reader = csv.reader(fhandle, delimiter="\t")
+    for line in reader:
+        times.append([float(line[0]), float(line[1])])
+        labels.append(line[2])
+        confidence.append(1.0) # There is no confidence annotation in MAVD-traffic v0.1.0
+
+    events_data = annotations.Events(np.array(times), labels, np.array(confidence))
+    return events_data
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The MAVD-traffic dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="mavd",
+            clip_class=Clip,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    @core.cached_property
+    def _metadata(self):
+
+        splits = ["train", "validate", "test"]
+        expected_sizes = [24, 7, 16]
```

Comment:
I recommend you check this in the `make_index` script if you want, i.e. while creating the index make sure you load all the files in each split and the length of the index you generated is the one it should be.

We don't test number of files directly in the library since it would require having all data somewhere and it's a lot of space. The loaders' functions are tested with toy data, and then the `test_full_dataset` checks general functionality and local files validity against the index. So if your index has the amount of expected files that is tested as well. For each user, that will be tested when doing `mavd.validate()`. Hope this is clear, let me know otherwise!